okapi/core.php

```diff
@@ -1206,6 +1206,83 @@ public static function get_node_code()
     }
 
     /**
+     * Return the recommended okapi_base_url.
+     *
+     * This is the URL which we want all *new* client applications to use.
+     * OKAPI will suggest URLs with this prefix in various context, e.g. in all
+     * the dynamically generated docs.
+     *
+     * Also see `get_allowed_base_urls` method.
+     */
+    public static function get_recommended_base_url()
+    {
+        return Settings::get('SITE_URL')."okapi/";
+    }
+
+    /**
+     * Return a list of okapi_base_urls allowed to be used when calling OKAPI
+     * methods in this installation.
+     *
+     * Since issue #416, the "recommended" okapi_base_url is *not* the only one
+     * allowed (actually, there were more allowed before issue #416, but they
+     * weren't allowed "officially").
+     */
+    public static function get_allowed_base_urls()
+    {
+        /* Currently, there are no config settings which would let us allow
+         * to determine the proper values for this list. So, we need to have it
+         * hardcoded. (Perhaps we should move this to etc/installations.xml?
+         * But this wouldn't be efficient...) */
+
+        switch (self::get_node_code()) {
+            case 'OCPL':
+                $urls = array(
+                    "http://opencaching.pl/okapi/",
+                    "http://www.opencaching.pl/okapi/",

```

Comment:
@deg-pl, @andrixnet  - OCPL will need keep allowing both of these domains when clients are accessing OKAPI - no Apache redirection should be allowed for `/okapi/` path.